lab_4_summarization_textrank/README.md

```diff
@@ -0,0 +1,758 @@
+
+Выделите из текста, хранящегося в переменной `text`, предложения и закодируйте их. Это необходимо сделать в файле `start.py`.
+
+### Шаг 6. Представить текст в виде графа. Матрица схожести предложений
+
+В рамках [лабораторной работы №3](../lab_3_keywords_textrank/README.md) Вы реализовали класс `AdjacencyMatrixGraph`, позволяющий строить граф на основе совстречаемости токенов.
+Токены были вершинами, а связи между ними - рёбрами.
+
+В данной работе Вы модифицируете функционал этого класса, чтобы можно было работать с предложениями:
+* Вершинами будут предложения
+* Рёбрами будут значения похожести между предложения
+  * Расчёт похожести между предложениями будет рассмотрен в следующих шагах
+
+В этом шаге Вы будете реализовывать `SimilarityMatrix`.
+
+Матрица смежности представляет собой не что иное, как таблицу. В нашем случае незаполненная матрица выглядит вот так.
+
+|   |A   |B   |C   |D   |
+|---|---|---|---|---|
+|A   |   |   |   |   |
+|B   |   |   |   |   |
+|C   |   |   |   |   |
+|D   |   |   |   |   |
+
+В ней одинаковое количество столбцов и строк. При этом каждая строка и каждый столбец соответствуют одной какой-либо вершине графа. Например, на пересечении строки, соответствующей вершине А, и столбца, соответствующего вершине В, будет записано число, 
+отражающее значение похожести между этими вершинами, являющимися предложениями. Давайте посмотрим, как она может выглядеть в заполненном виде.
+
+|   | A   | B   | C | D   |
+|---|-----|-----|---|-----|
+| A | 1   | 0.2 | 0 | 0.3 |
+| B | 0.2 | 1   | 0 | 0.7 |
+| C | 0   | 0   | 1 | 0   |
+| D | 0.3 | 0.7 | 0 | 1   |
+
+Ниже описаны шаги реализации матрицы похожести.
+
+#### Шаг 6.1 Вычисление метрики похожести 
+
+Предобработанное предложение - последовательность из токенов. Существует [множество способов](https://aircconline.com/mlaij/V3N1/3116mlaij03.pdf) узнать, 
+насколько две последовательности элементов похожи друг на друга.
+
+В данной работе Вы реализуете _меру Жаккара_ (_Jaccard Similarity/Index_), которая позволяет понять, насколько два множества элементов похожи.
+
+Формула выглядит следующим образом:
+
+```math
+J(A, B) = \frac{|A\cap B|}{|A\cup B|}
+```
+
+Объяснение:
+* $A$ & $B$ - множества элементов
+* $|A\cap B|$ - количество элементов, которые являются общими для обоих множеств
+* $|A\cup B|$ - общее количество уникальных элементов в обоих множествах
+
+Пример:
+* Множество $A$ - ${0, 1, 2, 5, 6}$
+* Множество $B$ - ${0, 2, 3, 4, 5, 7, 9}$
+
+```math
+J(A, B) = \frac{|A\cap B|}{|A\cup B|} = \frac{|\{0, 2, 5\}|}{|\{0, 1, 2, 3, 4, 5, 6, 7, 9\}|} = \frac{3}{9} = 0.3(3)
+```
+
+Реализуйте функцию, рассчитывающую значение меры Жаккара для двух последовательностей элементов.
+
+Функция принимает на вход две последовательности элементов.
+
+Функция возвращает значение меры Жаккара.
+
+Интерфейс:
+```py
+def calculate_similarity(sequence: Sequence, other_sequence: Sequence) -> float:
```

Comment:
method before initializer?? seems to be erroneous from conceptual point of view

Reply:
if this is not the method of class then it should be clearly stated and maybe a separate step (not a substep in matrix implementation)